Today/Extensions/ReminderExtension.swift

```diff
@@ -0,0 +1,50 @@
+//
+//  Reminder.swift
+//  Today
+//
+//  Created by BrainX Technologies 11 on 12/08/2021.
+//
+
+import UIKit
+
+extension Reminder {
+    
+    // MARK: - Static Properties
+    
+    static let timeFormatter: DateFormatter = {
+        let timeFormatter = DateFormatter()
+        timeFormatter.dateStyle = .none
+        timeFormatter.timeStyle = .short
+        return timeFormatter
+    }()
+    static let futureDateFormatter: DateFormatter = {
+        let dateFormatter = DateFormatter()
+        dateFormatter.dateStyle = .medium
+        dateFormatter.timeStyle = .short
+        return dateFormatter
+    }()
+    static let todayDateFormatter: DateFormatter = {
+        let format = NSLocalizedString(Strings.todayDateFormatString, comment: Strings.formatStringForDatesOccuringTodayString)
+        let dateFormatter = DateFormatter()
+        dateFormatter.dateFormat = String(format: format, Strings.hourMinuteTimeFormatString)
+        return dateFormatter
+    }()
+    
+    // MARK: - Public Methods
+    
+    public func dueDateTimeText(for filter: ReminderListDataSource.Filter) -> String {
+        let isInToday = Locale.current.calendar.isDateInToday(dueDate)
+        switch filter {
+        case .today:
+            return Self.timeFormatter.string(from: dueDate)
+        case .future:
+            return Self.futureDateFormatter.string(from: dueDate)
+        case .all:
+            if isInToday {
+                return Self.todayDateFormatter.string(from: dueDate)
+            } else {
+                return Self.futureDateFormatter.string(from: dueDate)
+            }
+        }
```

Comment:
Can we do something to remove the duplicated return statements ?

Reply:
![image](https://user-images.githubusercontent.com/85882612/129529922-ca93a487-7127-4334-8895-6b57d9dcbbad.png)

Reply:
is this approach acceptable ?

Reply:
But still these statements are repeating.
todayDateFormatter.string(from: dueDate)
futureDateFormatter.string(from: dueDate)

Reply:
Another way coming to my mind is to define properties for statements. 
Like this:
![image](https://user-images.githubusercontent.com/85882612/129562634-9a650f9a-5849-4379-b8d8-adf1110a60b9.png)

Reply:
Another way i thought of is this, but its giving me warning...
![image](https://user-images.githubusercontent.com/85882612/129564366-302bd708-1290-478b-87ff-ed5b6fa3d31a.png)

Reply:
How about his one ?
![image](https://user-images.githubusercontent.com/85882612/129564851-d5ea5059-c135-423b-9586-a225ff15dd2d.png)